activation/atxdb.go

```diff
@@ -513,8 +521,10 @@ func (db *DB) storeAtxUnlocked(atx *types.ActivationTx) error {
 	if err != nil {
 		return err
 	}
-	err = db.atxs.Put(getAtxBodyKey(atx.ID()), atxBodyBytes)
-	if err != nil {
+	if err = db.atxs.Put(getAtxBodyKey(atx.ID()), atxBodyBytes); err != nil {
+		return err
+	}
+	if err = db.atxs.Put(getAtxCoinbaseKey(atx), atx.ID().Bytes()); err != nil {
```

Comment:
There's a series of three `db.atxs.Put` calls here. Are these atomic? If so, where is this guaranteed? What happens if one of these succeeds and one fails?

Reply:
good catch - I've changed them to use a batch write instead. Also included more precise error messages to narrow down which put throws an error when one fails